pkg/deploy/controller/deployment/controller_test.go

```diff
@@ -651,6 +651,72 @@ func TestHandle_deployerPodDisappeared(t *testing.T) {
 	}
 }
 
+func expectMapContains(t *testing.T, exists, expected map[string]string, what string) {
+	if expected == nil {
+		return
+	}
+	for k, v := range expected {
+		value, ok := exists[k]
+		if ok && value != v {
+			t.Errorf("expected %s[%s]=%s, got %s", what, k, v, value)
+		} else if !ok {
+			t.Errorf("expected %s %s: not present", what, k)
+		}
+	}
+}
+
+func TestDeployerCustomLabelsAndAnnotations(t *testing.T) {
+	controller := &DeploymentController{
+		decodeConfig: func(deployment *kapi.ReplicationController) (*deployapi.DeploymentConfig, error) {
+			return deployutil.DecodeDeploymentConfig(deployment, api.Codec)
+		},
+		podClient: &podClientImpl{
+			createPodFunc: func(namespace string, pod *kapi.Pod) (*kapi.Pod, error) {
+				return pod, nil
+			},
+		},
+		makeContainer: func(strategy *deployapi.DeploymentStrategy) (*kapi.Container, error) {
+			return okContainer(), nil
+		},
+	}
+
+	testCases := []struct {
+		strategy     deployapi.DeploymentStrategy
```

Comment:
table tests are hard to debug when they start fail and you have a lot:) please add a name field here and plug it at the start of every `Errorf` like `t.Errorf("%s: label %s expected ..", test.name, test.other)`